zbus/src/address/transport.rs

```diff
@@ -0,0 +1,104 @@
+//! D-Bus supported transports.
+
+use std::fmt;
+
+use super::{
+    address::{KeyValFmt, KeyValFmtAdd},
+    percent, Address, Error, Result,
+};
+
+mod autolaunch;
+pub use autolaunch::{Autolaunch, AutolaunchScope};
+
+mod launchd;
+pub use launchd::Launchd;
+
+mod nonce_tcp;
+pub use nonce_tcp::NonceTcp;
+
+mod systemd;
+pub use systemd::Systemd;
+
+mod tcp;
+pub use tcp::{Tcp, TcpFamily};
+
+mod unix;
+pub use unix::{Unix, UnixAddrKind};
+
+mod unixexec;
+pub use unixexec::Unixexec;
+
+#[cfg(feature = "vsock_address")]
+mod vsock;
+#[cfg(feature = "vsock_address")]
+pub use vsock::Vsock;
+
+/// A D-Bus transport.
+#[derive(Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum Transport<'a> {
+    Unix(unix::Unix<'a>),
+    Launchd(launchd::Launchd<'a>),
```

Comment:
I understand why you originally decided to make all variants available on all platforms/targets, I can't help but wonder that on a practical level, this will only make the compiler do more work (and also add to that binary size, which we're already infamous for #304) while not giving any real benefit, given that use of address API without the actual transport, is at moment completely hypothetical. Moreover, you've not feature-gated the non-parsing code so it's not even possible to only use the address parsing API after this PR.

Thinking about this now (together with my other comment here), I remember that I had very good reasons for most of my decisions for the address API in 4.0. They were mostly driven by practical needs rather than theoretical ones.

Reply:
We are repeating ourself, I thought you already agreed (like many other subjects). The address parsing job is not specific to a platform. We could add more features similar to "vsock_address" if you believe it will help with resulting binary size.

Reply:
> We are repeating ourself,

What's new? :)

> I thought you already agreed (like many other subjects). The address parsing job is not specific to a platform.

As I wrote, it sounds good and correct in theory and that's why I agreed. I'm allowed to have more thoughts that can result in a different conclusion after agreeing. This is a pretty intrusive change with a lot of aspects to it and I've been failing to find one big chunk of time to properly analyze all aspects of it in one go.

I really shouldn't need to explain myself for changing my mind and having doubts about things I agreed with before. If my conclusions are final, we wouldn't have this PR since I told you multiple times to give up on this before. :)

> the address parsing job is not specific to a platform.

Repeating your assertion doesn't help. Please address my concerns above.

> We could add more features similar to "vsock_address" if you believe it will help with resulting binary size.

I don't think that's a good solution as it pushes the burdon on the user to disable those features, as they'll have to be the default if we want all of them supported on all platforms. This is easily automatable so we should and disregard a completely hypothetical advantage.

Reply:
My goal was to make a standalone fully compliant address parser. Now, slowly making zbus-specific this is becoming a thing that diverges from the original goal. You are free to make it more specific in further changes. It will make it harder to reuse outside of zbus..

Reply:
> My goal was to make a standalone fully compliant address parser.

Yeah, "was" being the keyword. I pointed out above, that this PR doesn't enable this use case so that goal hasn't succeeded yet. Also it's not something I care about, for the reasons I've described very clearly in detail already.

> It will make it harder to reuse outside of zbus

Not sure why you still keep bringing that up as some very compelling use case, while you've constantly failed to point to any actual possible users of this. You did so much work and had such long conversation to satisfy a use case that has a very high chance of never realising. :cry:

> You are free to make it more specific in further changes.

I'm sorry but I'm not going to do the work for you. I'm perfectly happy not merging this PR. You need to either address on my concerns or address them before this PR is merged.

---

Kindly stop resolving conversations unilaterally w/o addressing the concerns.

Reply:
added #[cfg] attributes where possible.